### PR TITLE
Make JDK 17 a default

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by findinpath** [comment]
I'm just wondering, how do we make sure that the images do start with the changes we are doing before making a new docker release?

**Review by kokosing — COMMENTED** [review]
How to test it with Trino? It would be great to make sure we are still able to run all tests in Trino after releasing these images.

**Comment by wendigo** [comment]
@kokosing it would be possible to test it if we were releasing snapshots. But since we are not, we can rely only on manual testing and/or testing after a release

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
I would test it manually, at least few product tests that are using TLS or KRB.

**Comment by hashhar** [comment]
Since the previous version includes everything except this PR I think we can merge this (after some manual tests) and perform a release. That way we always have a clean version of docker-images we can use in Trino if this fails.

File: testing/spark3.0-iceberg/Dockerfile [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM testing/centos7-oj11:unlabelled
+FROM testing/centos7-oj17:unlabelled
```
Comment (findinpath) [inline testing/spark3.0-iceberg/Dockerfile:13]: `oj` does it come from open jdk?

**Review by findinpath — APPROVED** [review]

**Comment by findinpath** [comment]
The commit message is a bit misleading:

```
Make JDK 17 a default
``` 


Maybe "Use Java 17 runtime" fits more the purpose of the commit.

File: testing/spark3.0-iceberg/Dockerfile (hunk 1 above)
Comment (wendigo) [reply testing/spark3.0-iceberg/Dockerfile:13]: openjdk

Comment (wendigo) [reply testing/spark3.0-iceberg/Dockerfile:13]: azul's builds are based on openjdk

**Review by hashhar — APPROVED** [review]

**Pull request merged**